src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -325,14 +325,42 @@ public void runFinalization() {
      * been cleared, either by the program or by the garbage collector, then
      * this method returns {@code null}.
      *
+     * @apiNote
+     * This method returns a strong reference to the referent. This may cause
+     * the garbage collector to treat it as strongly reachable until some later
+     * collection cycle.  The {@link #refersTo(Object) refersTo} method can be
+     * used to avoid such strengthening when testing whether some object is
+     * the referent of a reference object; that is, use {@code ref.refersTo(obj)}
+     * rather than {@code ref.get() == obj}.
+     *
      * @return   The object to which this reference refers, or
      *           {@code null} if this reference object has been cleared
+     * @see refersTo
      */
     @IntrinsicCandidate
     public T get() {
         return this.referent;
     }
 
+    /**
+     * Tests if this reference object refers to {@code obj}.  If {@code obj} is
+     * {@code null}, this method returns {@code true} if this reference object
+     * refers to {@code null} or has been cleared.
```

Comment:
Is there actually a distinction between between clearing and referring to null?

Reply:
A (not very useful) Reference can be created with a null referent.